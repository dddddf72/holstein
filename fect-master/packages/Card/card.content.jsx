import { defineComponent } from 'vue'
import './card.content.less'

const CardContent = defineComponent({
  setup(props, { attrs, slots }) {
    return () => (
      <>
        <div className={'fect-card-content'}>{slots && slots.default()}</div>
      </>
    )
  },
})

export default CardContent
