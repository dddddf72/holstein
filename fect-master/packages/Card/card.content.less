.fect-card-content {
  width: 100%;
  padding: 16pt;
  box-sizing: border-box;
  & *:first-child {
    margin-top: 0;
  }
  & *:last-child {
    margin-bottom: 0;
  }
}
