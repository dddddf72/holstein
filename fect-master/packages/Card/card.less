.fect-card {
  margin: 0;
  padding: 0;
  width: 100%;
  transition: all 0.2s ease;
  box-sizing: border-box;
  border-radius: 5px;
  border: 1px solid var(--accents-2);
}

.fect-card {
  &.hoverable {
    // box-shadow: 0 5px 10px rgba(0, 0, 0, 0.12);
    &:hover {
      box-shadow: 0 8px 30px rgba(0, 0, 0, 0.12);
    }
  }
  &.shadow {
    box-shadow: 0 5px 10px rgba(0, 0, 0, 0.12);
    border-color: transparent;
    // &:hover {
    //   box-shadow: 0 8px 30px rgba(0, 0, 0, 0.12);
    // }
  }
}
