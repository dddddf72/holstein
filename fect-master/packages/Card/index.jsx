import { computed, toRefs } from 'vue'
import { createNameSpace } from '../utils'
import CardContent from './card.content'

const [createComponent] = createNameSpace('Card')
import './card.less'

export default createComponent({
  props: {
    hoverable: Boolean,
    shadow: Boolean,
  },
  setup(props, { attrs, slots, emit }) {
    const { hoverable, shadow } = toRefs(props)

    const clacClass = computed(() => {
      let str = ''
      shadow.value && (str += ' shadow')
      hoverable.value && (str += ' hoverable')
      return str.trim()
    })

    return () => (
      <>
        <div
          {...attrs}
          className={`fect-card ${clacClass.value} ${
            attrs.class ? attrs.class : ''
          }`}
          style={attrs.style ? attrs.style : false}
        >
          <CardContent>{slots.default?.()}</CardContent>
        </div>
      </>
    )
  },
})
