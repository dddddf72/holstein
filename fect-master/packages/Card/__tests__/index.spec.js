import Card from '../index'
import { mount } from '@vue/test-utils'

describe('Card', () => {
  it('should be render as element', () => {
    const card = mount(Card)

    expect(card.html()).toMatchSnapshot()
  })

  it('should be support shdow and hoverable', () => {
    const wrapper = mount({
      setup() {},
      render(h) {
        return (
          <>
            <Card hoverable>hoverable Card</Card>
            <Card shadow>shadow Card</Card>
            <Card>hoverable and shadow Card</Card>
          </>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })
})
