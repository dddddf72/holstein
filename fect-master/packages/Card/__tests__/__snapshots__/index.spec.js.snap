// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should be render as element 1`] = `
<div class="fect-card  ">
  <div class="fect-card-content">
    <!---->
  </div>
</div>
`;

exports[`Card should be support shdow and hoverable 1`] = `
<div class="fect-card hoverable ">
  <div class="fect-card-content">hoverable Card</div>
</div>
<div class="fect-card shadow ">
  <div class="fect-card-content">shadow Card</div>
</div>
<div class="fect-card  ">
  <div class="fect-card-content">hoverable and shadow Card</div>
</div>
`;
